client/my-sites/people/edit-team-member-form/edit-user-form.jsx

```diff
@@ -98,10 +98,14 @@ class EditUserForm extends Component {
 		}
 
 		/*
-		 * If this is not a Jetpack site and the current user is not viewing their own profile,
+		 * If the current user is not viewing their own profile,
 		 * the user should not be able to edit the site owner's details.
 		 */
-		if ( ! isJetpack && user.ID === siteOwner && user.ID !== currentUser.ID ) {
+		if (
+			( ( ! isJetpack && user.ID === siteOwner ) ||
+				( isJetpack && user.linked_user_ID === siteOwner ) ) &&
+			user.ID !== currentUser.ID
+		) {
```

Comment:
It would be great to encapsulate this logic to a function, and write some tests around the function.

Given there are multiple scenarios covered by this conditional, tests would explicitly document the behavior for each scenario.

Reply:
Having the server-side tests, I'm thinking if we can skip writing tests for Calypso UX at this point. Currently, the conditions are documented via docblock comments and are written similarly to other conditions in this component.

Another thing is that we don't have any tests for this form yet. If we want to write them, we need to a) extract the current condition to separate component to avoid duplication and then write tests or b) write tests on the component level (mock data, render, check what it renders). I'm not sure if the effort would be reasonable now.

Reply:
@wojtekn I'm fine with this assessment.